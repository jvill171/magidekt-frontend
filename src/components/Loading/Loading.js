
import "./Loading.scss"


const Loading = () =>{

    return(
        <h1 className="Loading">
          <i className="ms ms-g"></i>
          <i className="ms ms-u"></i>
          <i className="ms ms-r"></i>
          <i className="ms ms-w"></i>
          <i className="ms ms-b"></i>
        </h1>
    )
}

export default Loading;