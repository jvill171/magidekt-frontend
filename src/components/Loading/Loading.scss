
/* Define the animation for jumping */
@keyframes jump {
    0%, 60% {
      transform: translateY(0); /* Initial and final position (no translation) */
    }
    30% {
      transform: translateY(-10px); /* Move up by 20 pixels */
    }
}

.Loading{
    .ms{
        border-radius: 50%;      /* Rounded border */
        padding: calc(2em / 15); /* 2/15 ths of parent elment font size*/
        margin: calc(1em/15);     /* 1/15 ths of parent elment font size*/
        box-shadow: calc(-1em/15) calc(1em/15) 0 rgba(0,0,0,0.85);
        border: 1px solid black;
        color:black;
        animation: jump 1.5s ease-in-out infinite; // Adjust the duration as needed
      }

      
    .ms-g{
        background-color: #9cd4ad;
        animation-delay: 0.1s;
    }
    .ms-u{
        background-color: #a9e1fa;
        animation-delay: 0.2s;
    }
    .ms-r{
        background-color: #fbaa8f;
        animation-delay: 0.3s;
    }
    .ms-w{
        background-color: #fefcd6;
        animation-delay: 0.4s;
    }
    .ms-b{
        background-color: #cdc2c0;
        animation-delay: 0.5s;
    }

}
