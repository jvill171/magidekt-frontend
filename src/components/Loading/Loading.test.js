import React from 'react';
import { render } from '@testing-library/react';
import Loading from './Loading';

//Smoke test
test('renders Loading component without crashing', () => {
  render( <Loading /> );
});

//Snapshot test
test('Loading component matches snapshot', () => {
  const { asFragment } = render( <Loading /> );

  expect(asFragment()).toMatchSnapshot();
});
