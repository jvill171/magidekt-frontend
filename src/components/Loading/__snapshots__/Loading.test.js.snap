// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component matches snapshot 1`] = `
<DocumentFragment>
  <h1
    class="Loading"
  >
    <i
      class="ms ms-g"
    />
    <i
      class="ms ms-u"
    />
    <i
      class="ms ms-r"
    />
    <i
      class="ms ms-w"
    />
    <i
      class="ms ms-b"
    />
  </h1>
</DocumentFragment>
`;
